util/jsonpull.c

```diff
@@ -0,0 +1,970 @@
+
+  switch (parser->last_read_char)
+    {
+    case '"':
+      if (gvm_json_pull_parse_string (parser, event, &key_cjson))
+        return 1;
+      key_str = g_strdup (key_cjson->valuestring);
+      cJSON_free (key_cjson);
+
+      // Expect colon:
+      if (gvm_json_pull_skip_space (parser, event, FALSE))
+        {
+          g_free (key_str);
+          return 1;
+        }
+      if (parser->last_read_char != ':')
+        {
+          event->type = GVM_JSON_PULL_EVENT_ERROR;
+          event->error_message = g_strdup_printf ("expected colon");
+          g_free (key_str);
+          return 1;
+        }
+      gvm_json_pull_parser_next_char (parser);
+
+      path_elem = g_queue_peek_tail (parser->path);
+      g_free (path_elem->key);
+      path_elem->key = key_str;
+      parser->expect = GVM_JSON_PULL_EXPECT_VALUE;
+
+      break;
+    case '}':
+      event->type = GVM_JSON_PULL_EVENT_OBJECT_END;
+      event->value = NULL;
+      gvm_json_pull_path_elem_free (g_queue_pop_tail (parser->path));
+      PARSE_VALUE_NEXT_EXPECT
+      gvm_json_pull_parser_next_char (parser);
+      break;
+    case ']':
+      event->type = GVM_JSON_PULL_EVENT_ERROR;
+      event->error_message = g_strdup ("unexpected closing square bracket");
+      return 1;
+    default:
+      event->type = GVM_JSON_PULL_EVENT_ERROR;
+      event->error_message = g_strdup ("unexpected character");
+      return 1;
+    }
+
+  return 0;
+}
+
+/**
+ * @brief Handles the case that a comma is expected in a JSON pull parser.
+ *
+ * This will continue the parsing until a comma or the end of the
+ * current array/object was reached or an error occurred.
+ *
+ * @param[in]  parser   Parser to process
+ * @param[in]  event    Event data to set error or end of object status in
+ *
+ * @return 1 if an error occurred, 0 otherwise
+ */
+static int
+gvm_json_pull_parse_comma (gvm_json_pull_parser_t *parser,
+                           gvm_json_pull_event_t *event)
+{
+  if (gvm_json_pull_skip_space (parser, event, FALSE))
+    return 1;
+
+  gvm_json_path_elem_t *path_elem = NULL;
+  switch (parser->last_read_char)
+    {
+    case ',':
+      path_elem = g_queue_peek_tail (parser->path);
+      path_elem->index++;
+      if (path_elem->parent_type == GVM_JSON_PULL_CONTAINER_OBJECT)
+        parser->expect = GVM_JSON_PULL_EXPECT_KEY;
+      else
+        parser->expect = GVM_JSON_PULL_EXPECT_VALUE;
+      gvm_json_pull_parser_next_char (parser);
+      break;
+    case ']':
+      path_elem = g_queue_peek_tail (parser->path);
+      if (path_elem == NULL
+          || path_elem->parent_type != GVM_JSON_PULL_CONTAINER_ARRAY)
+        {
+          event->type = GVM_JSON_PULL_EVENT_ERROR;
+          event->error_message = g_strdup ("unexpected closing square bracket");
+          return 1;
+        }
+      event->type = GVM_JSON_PULL_EVENT_ARRAY_END;
+      event->value = NULL;
+      gvm_json_pull_path_elem_free (g_queue_pop_tail (parser->path));
+      PARSE_VALUE_NEXT_EXPECT
```

Comment:
```suggestion
      parse_value_next_expect (parser);
```
And change others below.